wp-content/themes/twentyseventeen-wp20/style.css

```
@@ -863,94 +864,157 @@ body.page .entry-header {
 }
 
 /*
- * What's on
+ * What's On - Builds on top of plugin's default styles.
  */
 
-#wp20-events-map {
-	position: relative;
-	height: 500px;
-	width: 80vw;
-	left: 50%;
-	right: 50%;
-	margin-left: -40vw;
-	margin-right: -40vw;
-	margin-top: 2em;
-	margin-bottom: 2em;
-	background-color: #AADAFF;
+body.page-slug-whats-on.page-two-column:not(.archive) #primary .entry-header,
+body.page-slug-whats-on.page-two-column:not(.archive) #primary .entry-content {
+	float: none;
+	display: block;
+	width: 100%;
+}
+
+body.page-slug-whats-on.page-two-column:not(.archive) #primary .entry-header {
+	display: flex;
+	justify-content: space-between;
+	align-items: flex-end;
 }
 
 @media screen and ( max-width: 48em ) {
-	body.page.page-slug-whats-on .entry-header {
+	body.page-slug-whats-on.page-two-column:not(.archive) #primary .entry-header {
+		display: block;
 		margin-bottom: var(--wp20--spacing--small);
 	}
+}
 
-	#wp20-events-map {
-		width: 100%;
-		margin: 2em auto 1em;
-		left: auto;
-		right: auto;
+body.page-slug-whats-on h1.entry-title {
+	width: 639px;
+	font-size: clamp( 20px, 2.4vw, 30px);
+	margin-right: var(--wp20--spacing--medium);
+	margin-bottom: 0;
+}
+@media screen and ( max-width: 48em ) {
+	body.page-slug-whats-on h1.entry-title {
+		width: auto;
+		margin-bottom: var(--wp20--spacing--small);
 	}
 }
 
-@media screen and ( min-width: 769px ) {
-	body.page.page-slug-whats-on .entry-header .entry-title {
-		width: 55%;
+#wp20-events-map {
+	background-color: #AADAFF;
+	margin: 0 auto 40px;
+}
+@media screen and ( max-width: 48em ) {
+	#wp20-events-map {
+		position: relative;
+		width: 100vw;
+		left: calc( -1 * var(--wp20--spacing--edge-space) );
 	}
 }
 
 #wp20-events-filter {
-	margin-bottom: 3em;
-	text-align: center;
+	position: relative;
+	display: flex;
+	flex-wrap: nowrap;
+	box-sizing: border-box;
+	min-width: 325px;
+	margin-bottom: 0;
+	padding-right: 0;
 }
 
-#wp20-events-filter input[type="text"] {
-	display: inline;
-	width: auto;
-	margin-left: .5em;
+#wp20-events-filter label {
+	margin-right: 30px;
+	margin-bottom: 0;
+	padding: 0.7em 0; /* Must match `.wp20-events-query` (inherited from TwentySeventeen. */
+}
+
+#wp20-events-filter label span {
+	font-size: 13px;
+}
+
+#wp20-events-filter .wp20-events-query {
+	flex-shrink: 0;
+	flex-grow: 1;
+	flex-basis: 200px;
+	border-radius: 2px;
+	border-color: var(--wp20--color--grey-0);
+	font-size: 12px;
 }
 
-#wp20-events-filter input[type="text"]::placeholder {
+#wp20-events-filter .wp20-events-query::placeholder {
 	opacity: 0.5;
 }
 
-.rtl #wp20-events-filter input[type="text"] {
-	margin-left: 0;
-	margin-right: .5em;
+#wp20-events-query-mobile {
+	display: none;
+}
```

The styles are written mobile first, so it would make sense to reverse this

```suggestion
#wp20-events-query-desktop {
	display: none;
}
```

Just need to be careful when doing the switch that the right input is displayed at 768px. I assume that's the mobile one, so the media query for showing desktop will be `min-width: 769px`, rather than `min-width: 48em`

---

Ah, the ones I'd encountered seemed to be desktop-first, so I did it that way here for consistency. Looking through more comprehensively, though, it looks like it's a mix of mobile-first and desktop-first.

Do you think this should be changed to mobile-first? I don't have a strong opinion.